queue_job/models/queue_job.py

```diff
@@ -127,38 +128,43 @@ class QueueJob(models.Model):
     worker_pid = fields.Integer(readonly=True)
 
     def init(self):
+        cr = self.env.cr
         index_1 = "queue_job_identity_key_state_partial_index"
         index_2 = "queue_job_channel_date_done_date_created_index"
-        if not index_exists(self._cr, index_1):
-            # Used by Job.job_record_with_same_identity_key
-            self._cr.execute(
-                "CREATE INDEX queue_job_identity_key_state_partial_index "
-                "ON queue_job (identity_key) WHERE state in ('pending', "
-                "'enqueued', 'wait_dependencies') AND identity_key IS NOT NULL;"
-            )
-        if not index_exists(self._cr, index_2):
-            # Used by <queue.job>.autovacuum
-            self._cr.execute(
-                "CREATE INDEX queue_job_channel_date_done_date_created_index "
-                "ON queue_job (channel, date_done, date_created);"
-            )
+        # Used by Job.job_record_with_same_identity_key
+        create_index(
+            cr,
+            index_1,
+            "queue_job",
+            ["identity_key"],
+            where=(
+                "state in ('pending','enqueued','wait_dependencies') "
+                "AND identity_key IS NOT NULL"
+            ),
+            comment=("Queue Job: partial index for identity_key on active states"),
+        )
+        # Used by <queue.job>.autovacuum
+        create_index(
+            cr,
+            index_2,
+            "queue_job",
+            ["channel", "date_done", "date_created"],
+            comment="Queue Job: index to accelerate autovacuum",
+        )
 
     @api.depends("dependencies")
     def _compute_dependency_graph(self):
-        jobs_groups = self.env["queue.job"].read_group(
-            [
-                (
-                    "graph_uuid",
-                    "in",
-                    [uuid for uuid in self.mapped("graph_uuid") if uuid],
-                )
-            ],
-            ["graph_uuid", "ids:array_agg(id)"],
-            ["graph_uuid"],
-        )
-        ids_per_graph_uuid = {
-            group["graph_uuid"]: group["ids"] for group in jobs_groups
-        }
+        uuids = [uuid for uuid in self.mapped("graph_uuid") if uuid]
+        ids_per_graph_uuid = {}
+        if uuids:
+            rows = self.env["queue.job"]._read_group(
+                [("graph_uuid", "in", uuids)],
+                groupby=["graph_uuid"],
+                aggregates=["id:recordset"],
+            )
+            # rows -> list of tuples: (graph_uuid, recordset)
+            for graph_uuid, recs in rows:
+                ids_per_graph_uuid[graph_uuid] = recs.ids
```

Comment:
```suggestion
            ids_per_graph_uuid = dict(self.env["queue.job"]._read_group(
                [("graph_uuid", "in", graph_uuids)],
                groupby=["graph_uuid"],
                aggregates=["id:array_agg"],
            ))
```

This is simpler and more efficient, as we don't need the recordsets.

Reply:
Switched to `_read_group(..., aggregates=["id:array_agg"])` and `dict(...)` to avoid building recordsets just to get ids. Thanks!